packages/vdom-extension/src/index.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { JupyterFrontEnd, ILayoutRestorer } from '@jupyterlab/application';
+
+import { InstanceTracker } from '@jupyterlab/apputils';
+
+import { MimeDocumentFactory, MimeDocument } from '@jupyterlab/docregistry';
+
+import { INotebookTracker } from '@jupyterlab/notebook';
+
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+
+import { RenderedVDOM } from './widget';
+
+/**
+ * The CSS class for a VDOM icon.
+ */
+const CSS_ICON_CLASS = 'jp-MaterialIcon jp-VDOMIcon';
+
+/**
+ * The MIME type for VDOM.
+ */
+export const MIME_TYPE = 'application/vdom.v1+json';
+
+/**
+ * The name of the factory that creates VDOM widgets.
+ */
+const FACTORY_NAME = 'VDOM';
+
+const plugin = {
+  id: '@jupyterlab/vdom-extension:factory',
+  requires: [IRenderMimeRegistry, INotebookTracker, ILayoutRestorer],
+  autoStart: true,
+  activate: (
+    app: JupyterFrontEnd,
+    rendermime: IRenderMimeRegistry,
+    notebooks: INotebookTracker,
+    restorer: ILayoutRestorer
+  ) => {
+    const tracker = new InstanceTracker<MimeDocument>({
```

Comment:
Ideally, I'd like to use [IMimeDocumentTracker](https://github.com/jupyterlab/jupyterlab/blob/master/packages/application/src/mimerenderers.ts#L47) which is used to track all other mimerender extensions, but this works.

Reply:
Yeah, that is really oriented around mime renderer extensions. Since this is no longer one of those, I think this is the correct approach.